### DISCUSSION TITLE
The UTILS_HAS_THREADING=0 can't work on Android?

### DISCUSSION THREAD

**Discussion by dugufly** [google/filament#4719] [Q&A]
If I set UTILS_HAS_THREADING to 0, it can  work well on window , but it cannot run with the sample-gltf-viewer on Android. What adaptation needs to be done on Android?

**Comment by romainguy** [comment]
This is only something we use for our WebGL builds. Why do you want to turn off multithreading?

**Reply by dugufly** [reply]
My rendering logic is placed in GLSurfaceView. I just want to treat the filament as one of my rendering steps. There are other rendering logics before and after the filament. I want to input a texture to the filament and output a texture. If the filament has a separate rendering thread, data synchronization will waste performance.

**Reply by romainguy** [reply]
You should really avoid this, first because Filament does little internal synchronization and when it does it’s well optimized specifically on Android. In addition you risk messing with the internal GL state by accident. I highly recommend you leave this setting alone. Also note that we will remove this setting once we can use wasm multi threading in all browsers.

**Reply by pixelflinger** [reply]
There is no way to make what you describe work. Even in single threaded mode, filament cannot share a GLES context.

Unfortunately, filament is not designed to be embedded, in this way, in another engine.

Moreover, doing things this way would assume that filament is using GLES, which won't be necessarily true in the future.
